### PR TITLE
feat: plat-5537 authenticated api updates from from smalti

### PR DESCRIPTION
**Pull request by malcyL**
- https://github.com/talis/platform/issues/5537

This PR pulls in modifications to the Authenticated API Construct made in the lti microservice project. That project did not start by using the construct form this library, instead recreating it in it's own repo. It followed the design of the AuthenticatedAPI construct from this repo, but made improvements to it - hence this PR to retro fit those improvements before it is used elsewhere.

The improvements are:

- The AuthenticatedAPI no longer creates the lambda's inside the construct. Instead the lambda are created outside and passed in. This allows for permissions to be added to the lambda in the standard cdk was rather than having to pass policy statements into the api construct.
- The AuthenticatedAPI now creates a domain name
- An extra cloudwatch alarm has been added to report on lmabda errors.
- We export the name of the API to the console for ease of use
- The Authorizer function is now typescript

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
     "integration-test": "npm run build && npm run jest-integration-test",
     "jest-integration-test": "jest test/integration/",
     "lint:prettier": "prettier --check .",
-    "lint:eslint": "eslint . --ext .js,.ts",
+    "lint:eslint": "eslint . --ext .ts",
```
Comment (stephenrob) [inline package.json:18]: Removing `js` from the linter is because we've moved the authorizer to typescript ?

**Review by stephenrob — APPROVED** [review]
Looks good - Doesn't seem to be any issues when its been moved from SMALTI

File: package.json (hunk 1 above)
Comment (malcyL) [reply package.json:18]: Yes. This was there to lint the original `authorizer.js` file. We don't want to lint it anymore, now it's the output of compiling the `.ts` file.

**Pull request merged**